scripts/api.mjs

```diff
@@ -0,0 +1,36 @@
+/**
+ * Calls the api with the parameters given
+ * @param {string} method HTML method to be used when calling the api 
+ * @param {object} body Object to post to the api, if body is not required it should be sett to null
+ * @param {string} url Api endpoint to call
+ * @returns Parsed response from api
+ */
+export default async function apiCall(method, body, url){
+  const options = {
+    method: method,
+    headers:{
+      "Content-Type": "application/json",
+    },
+  };
+
+  if(window.sessionStorage.getItem("key")){
+    const key = JSON.parse(window.sessionStorage.getItem("key"));
```

Comment:
```suggestion
    const key = JSON.parse(window.localStorage.getItem("key"));
```

Reply:
Local storage persists across multiple sessions. Session storage will log you out each time you start a new session which is not recommended.

Reply:
I've now changed to storing the user object in local storage.

The new post function is also functioning now, displaying the new post after its posted without needing to manually update the page